apps/base-docs/tutorials/docs/4_smart-wallet-and-eoa-with-onchainkit.md

```diff
@@ -0,0 +1,256 @@
+import { connectorsForWallets } from '@rainbow-me/rainbowkit';
+
+const connectors = connectorsForWallets(
+  [
+    {
+      groupName: 'Recommended Wallet',
+      wallets: [coinbaseWallet],
+    },
+    {
+      groupName: 'Other Wallets',
+      wallets: [rainbowWallet, metaMaskWallet],
+    },
+  ],
+  {
+    appName: 'onchainkit',
+    projectId: '<YOUR WALLET CONNECT ID>',
+  },
+);
+
+export const wagmiConfig = createConfig({
+  chains: [baseSepolia],
+  // turn off injected provider discovery
+  multiInjectedProviderDiscovery: false,
+  connectors,
+  ssr: true,
+  transports: {
+    [baseSepolia.id]: http(),
+  },
+});
+
+declare module 'wagmi' {
+  interface Register {
+    config: typeof wagmiConfig;
+  }
+}
+```
+
+Next, open `OnchainProviders.tsx`. Import and add the css for RainbowKit, and the context provider. You'll also need a css file from OnchainKit:
+
+```tsx
+'use client';
+import { OnchainKitProvider } from '@coinbase/onchainkit';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import type { ReactNode } from 'react';
+import { base } from 'viem/chains';
+import { WagmiProvider } from 'wagmi';
+import { NEXT_PUBLIC_CDP_API_KEY } from '../config';
+import { wagmiConfig } from '../wagmi';
+import { RainbowKitProvider } from '@rainbow-me/rainbowkit';
+
+import '@coinbase/onchainkit/styles.css';
+import '@rainbow-me/rainbowkit/styles.css';
+
+type Props = { children: ReactNode };
+
+const queryClient = new QueryClient();
+
+function OnchainProviders({ children }: Props) {
+  return (
+    <WagmiProvider config={wagmiConfig}>
+      <QueryClientProvider client={queryClient}>
+        <OnchainKitProvider apiKey={NEXT_PUBLIC_CDP_API_KEY} chain={base}>
+          <RainbowKitProvider modalSize="compact">{children}</RainbowKitProvider>
+        </OnchainKitProvider>
+      </QueryClientProvider>
+    </WagmiProvider>
+  );
+}
+
+export default OnchainProviders;
+```
+
+Finally, in `WalletComponents.tsx`, add `withWalletAggregator` to `<ConnectWallet>`:
+
+```tsx
+<Wallet>
+  <ConnectWallet withWalletAggregator>
+    <Avatar className="h-6 w-6" />
+    <Name />
+  </ConnectWallet>
+  {/* Code continues... */}
+</Wallet>
+```
+
+You've made changes that sometimes confuse the wallet connection, so clear site data, refresh the page, then try connecting again. Now, you'll get the RainbowKit modal with your list of wallets. You've solved the problem of allowing other types of wallets to connect to your app, but you'll find that the Coinbase wallet works as before - it will default to the EOA if a browser extension is present, but the smart wallet if it is not.
+
+## Tuning the Coinbase Wallet Connection
+
+:::danger
+
+If you've already connected with the site, the below won't work unless you clear site data! It will appear that the flag does nothing.
+
+:::
+
+With RainbowKit, you can force the connector to use the smart wallet or EOA with:
+
+```tsx
+coinbaseWallet.preference = 'smartWalletOnly';
```

Comment:
Where do I do this exactly?

I would expand the code.  ex: this gives me an error:

```
    {
      groupName: 'Recommended',
      wallets: [(coinbaseWallet.preference = 'smartWalletOnly')],
    },
    ```